src/lib/coda_lib/coda_lib.ml

```diff
@@ -789,16 +788,15 @@ module Make (Inputs : Inputs_intf) = struct
               (Linear_pipe.iter (Snark_pool.broadcasts snark_pool) ~f:(fun x ->
                    Net.broadcast_snark_pool_diff net x ;
                    Deferred.unit )) ;
-            let proposer_kps_r, proposer_kps_w =
-              Strict_pipe.(
-                create ~name:"propose_keypairs"
-                  (Buffered (`Capacity 1, `Overflow Drop_head)))
-            in
-            Strict_pipe.Writer.write proposer_kps_w
-              config.initial_propose_keypairs ;
             return
-              { propose_keypairs= proposer_kps_r
-              ; propose_keypairs_writer= proposer_kps_w
+              { propose_keypairs=
+                  Agent.create
+                    ~f:(fun kps ->
+                      Keypair.Set.to_list kps
```

Comment:
Small nit: can't you use `Keypair.Set.map`?

Reply:
Doesn't let you change set module types with `map`

Reply:
Sad